sweepbatcher/sweep_batch.go

```diff
@@ -1314,15 +1335,80 @@ func constructUnsignedTx(sweeps []sweep, address btcutil.Address,
 		feeForWeight++
 	}
 
+	// Add the batch transaction output, which excludes the fees paid to
+	// miners. Reduce the amount by the sum of change outputs, if any.
+	var sumChange int64
+	for _, change := range changeOutputs {
+		sumChange += change.Value
+	}
+
+	// Ensure that the batch amount is greater than the sum of change.
+	if batchAmt <= btcutil.Amount(sumChange) {
+		return nil, 0, 0, 0, fmt.Errorf("batch amount %v is less than "+
+			"the sum of change outputs %v", batchAmt,
+			btcutil.Amount(sumChange))
+	}
+
 	// Clamp the calculated fee to the max allowed fee amount for the batch.
-	fee := clampBatchFee(feeForWeight, batchAmt)
+	fee := clampBatchFee(feeForWeight, batchAmt-btcutil.Amount(sumChange))
 
-	// Add the batch transaction output, which excludes the fees paid to
-	// miners.
+	// Ensure that batch amount exceeds the sum of change outputs and the
+	// fee, and that it is also greater than dust limit for the main
+	// output.
+	if fee+btcutil.Amount(sumChange)+
+		lnwallet.DustLimitForSize(len(batchPkScript)) > batchAmt {
```

Comment:
`lnwallet.DustLimitForSize` calls `mempool.GetDustThreshold` under the hood (which we also use below).
Let's use `mempool.GetDustThreshold` in all places.
We can make a wrapper function like this to avoid copy-paste:
```go
func dustLimitForPkScript(pkscript []byte) btcutil.Amount
```
It would call `mempool.GetDustThreshold` under the hood. It only depends on pkscript, not on the value of TxOut.

And we will always use it in SB, e.g. in `checkSweepToAdd` to guarantee uniformity.

Reply:
I created function `dustLimitForPkScript` which we call in `checkSweepToAdd` to check dust size on optional change outputs of sweeps, and in `constructUnsignedTx`.